### ISSUE TITLE
Can't logout from Custom CCP with third-party-cookie phaseout

### ISSUE THREAD

**Issue by orihika0123** [amazon-connect/amazon-connect-streams#778]
I am working on a custom CCP using Streams v2.7.2, with the embedded CCP set to hidden. In order to handle third-party cookies as per the instructions on this [page](https://docs.aws.amazon.com/connect/latest/adminguide/admin-3pcookies.html), I have enabled Chrome's experimental feature test-third-party-cookie-phaseout. However, when attempting to logout from the CCP using the following code, but can't logout.

```
fetch("https://<xxxxxxxxx>/connect/logout", { credentials: 'include', mode: 'no-cors'})
```

It appears the request is being made without attaching any cookies.
![スクリーンショット 2023-10-04 11 27 26](https://github.com/amazon-connect/amazon-connect-streams/assets/34177860/81fb185b-df60-4b7b-baab-131a52e43ca4)

How can I resolve this issue to ensure a successful logout from the CCP?
Is the only solution to utilize first-party cookies for logging out by opening the logout URL in a separate window?

**Comment by MHamza8070** [comment]
@orihika0123  https://github.com/amazon-connect/amazon-connect-streams/issues/782 , I am trying to implement third party cookies phaseout and facing the issue shared in above line, can you please guide us how to implement this correctly.

**Comment by andywang219** [comment]
@orihika0123, please make sure that you are also triggering the `TERMINATE` event. Please see: https://github.com/amazon-connect/amazon-connect-streams/blob/master/Documentation.md#logging-out

**Comment by orihika0123** [comment]
@andywang219 Thank you for the advice!

I tried adding a TERMINATE event trigger as follows:
```
connect.core.initCCP(ccpDiv, {
    ccpUrl: `https://${callCenterDomain}.my.connect.aws/ccp-v2#/`,
    region: 'ap-northeast-1',
    loginPopup: true,
    softphone: {
      allowFramedSoftphone: false,
      disableRingtone: false,
    },
    storageAccess: {
      mode: "custom"
    }
  })

connect.core.initSoftphoneManager({allowFramedSoftphone: true})

const logout = () => {
  fetch(`https://${callCenterDomain}.my.connect.aws/logout`, { credentials: 'include', mode: 'no-cors'})
    .then(() => {
      connect.core.getEventBus().trigger(connect.EventType.TERMINATE);
    });
}
```

Unfortunately, even with the trigger added, I couldn't log out.

When executing the logout fetch:
- In the logout get process, all cookies are being filtered.
   The reason for filtering appears to be cross-site context?
![スクリーンショット 2023-10-24 11 54 29](https://github.com/amazon-connect/amazon-connect-streams/assets/34177860/d309255f-2de7-4972-8793-fbe15a91668f)
   
- After fetch and reloading the page, I'm not logged out from the CCP(= The CCP launches automatically without any login action).
   It seems that if I could delete the cookie from the browser, I would be able to log out. However, since the cookie has the httponly attribute, it seems I cannot delete it from within my code.

The environments I tested are:
Chrome versions: 117, 118, 119β
Experimental Flags: Storage Access API, Storage Access API permission UI, Test Third Party Cookie Phaseout all set to enabled.
![スクリーンショット 2023-10-24 12 04 47](https://github.com/amazon-connect/amazon-connect-streams/assets/34177860/328dea8a-b729-4f0f-871a-2fceaacb7fb3)

I believe there might be a need for a new method in Amazon Connect Streams to log out of the CCP.
Without this method, the workaround might be to open the logout URL in a separate window, which is visible to the user. I'd like to avoid this as it degrades the user experience.
